### PR TITLE
[WASM] Fix pointers for elt nested in manip aware content

### PR DESCRIPTION
**Pull request by dr1rrb**
GitHub Issue: fixes#https://github.com/unoplatform/uno/issues/2843

## Bugfix
Fix pointers event for elements that are nested in an element that has enabled manipulations.

## What is the current behavior?
When a manipulation stars on an element, it will implicitly capture the pointer to make sure to receive all updates even when pointer moves out of the range of this element. But this will prevent any nested child to receive pointer events (instead they will receive a 'pointerleave' on capture and a 'pointerenter' when releasing capture ... for mouse).

## What is the new behavior?
For implicit captures we use the `OriginalSource` (if possible) instead of the element that is actually capturing the pointer, so event will still bubble up from the lowest layer.

## PR Checklist
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/UIElement.PointerCapture.cs [hunk 1]
```diff
@@ -225,6 +243,12 @@ public bool ValidateAndUpdate(UIElement element, PointerRoutedEventArgs args, bo
 
 					return true;
 				}
+				else if (IsImplicitOnly)
+				{
+					// If the capture is implicit, we should not filter out events for child elements.
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/UIElement.PointerCapture.cs]:
```suggestion
					// If the capture is implicit, we should not filter out events for children elements.
```

File: src/Uno.UI/UI/Xaml/UIElement.PointerCapture.cs [hunk 2]
```diff
@@ -268,7 +292,7 @@ private void EnsureEffectiveCaptureState()
 
 					if (_nativeCaptureElement == null)
 					{
-						_nativeCaptureElement = _targets.Single().Key;
+						_nativeCaptureElement = _targets.Single().Value.NativeCaptureElement;
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/UIElement.PointerCapture.cs]: Guaranteed to have a value ?

File: src/Uno.UI/UI/Xaml/UIElement.PointerCapture.cs [hunk 3]
```diff
@@ -102,10 +108,15 @@ public bool TryAddTarget(UIElement element, PointerCaptureKind kind, PointerRout
 					{
 						return false;
 					}
+					else
+					{
+						// Add the new king to the target
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/UIElement.PointerCapture.cs]:
```suggestion
						// Add the new kind to the target
```

File: src/Uno.UI/UI/Xaml/UIElement.PointerCapture.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/UIElement.PointerCapture.cs]: Yes, otherwise I fallback to the `element` (so my fix doesn't apply)

**Comment by davidjohnoliver** [comment]
Seems like some tests are failing

image

**Pull request merged**